clients/java/signalr/src/main/java/HubConnection.java

```diff
@@ -55,13 +55,39 @@ public void stop(){
         connected = false;
     }
 
-    public void send(String method, Object... args) {
+    public void send(String method, Object... args) throws Exception {
         InvocationMessage invocationMessage = new InvocationMessage(method, args);
         String message = protocol.writeMessage(invocationMessage);
         transport.send(message);
     }
 
     public void On(String target, Action callback) {
-        handlers.put(target, callback);
+        ActionBase actionBase = new ActionBase(callback);
+        handlers.put(target, actionBase);
+    }
+
+    public <T1> void On(String target, Action1<T1> callback, Class<T1> param1) {
```

Comment:
Just so I understand, to call this I'd have to write:

```java
connection.on("Foo", (a, b) -> {
    // do stuff with a, b
}, String.class, SomeCustomType.class);
```

Correct? Type Erasure sucks 😢

Reply:
I should've just kept reading the PR ;P